pgmpy/tests/test_models/test_BayesianNetwork.py

```diff
@@ -1807,23 +1805,77 @@ def test_missing_scheme_sample(self):
         samples = self.con_model.simulate(n_samples=1000)
         self.assertFalse(samples.isnull().values.any())
 
-        samples = self.con_model.simulate(
-            n_samples=1000, include_missing=True, missing_prob=0.9
+        cpd = TabularCPD(
+            "Z*",
+            2,
+            [[0.5], [0.5]],
+        )
+        samples = self.con_model.simulate(n_samples=1000, missing_prob=cpd)
+        self.assertTrue(samples["Z"].isnull().values.any())
+        self.assertFalse(samples.drop(columns=["Z"]).isnull().values.any())
+
+        cpd = TabularCPD(
+            "Z*", 2, [[0.5, 0.5, 0.5, 0.5], [0.5, 0.5, 0.5, 0.5]], ["X", "Y"], [2, 2]
+        )
+        samples = self.con_model.simulate(n_samples=1000, missing_prob=cpd)
+        self.assertTrue(samples["Z"].isnull().values.any())
+        self.assertFalse(samples.drop(columns=["Z"]).isnull().values.any())
```

Comment:
Similarly, we can test the expected number of missing values with non MCAR cases. But as the number of missing values depends on some other variable now, we would need to take subsamples corresponding to the states of the parent variables, and then test for the expected number of missing values in that.